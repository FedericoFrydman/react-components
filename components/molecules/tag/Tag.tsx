import { forwardRef, HTMLProps, Ref } from 'react';
import cn from 'classnames';

import { Icon } from '../../atoms/icon/Icon';
import { Text } from '../../atoms/text/Text';
import './tag.scss';

export interface TagProps extends Omit<HTMLProps<HTMLDivElement>, 'css'> {
  /**
   * Text to display
   */
  text: string;
  /**
   * Additional data to display selected style
   */
  isSelected?: boolean;
  /**
   * Additional data that is dispatched with the tracking event.
   */
  onDismissible?: (e: React.MouseEvent) => void | null;
  /**
   * Additional data that is dispatched with the tracking event.
   */
  onSelected?: (e: React.MouseEvent) => void | null;
  /**
   * Custom classes for the component.
   */
  classNames?: cn.Argument;
  /**
   * The ref to the HTML DOM element
   */
  ref?: Ref<HTMLDivElement>;
}

const TagComponent = (
  { text, isSelected, onDismissible, onSelected, classNames = [], ...props }: TagProps,
  ref: TagProps['ref'],
) => {
  const classTag = 'lsux-tag';
  const cnTag = cn(
    classTag,
    {
      [`${classTag}--dismissible`]: onDismissible,
      [`${classTag}--selectable`]: !!onSelected,
      [`${classTag}--selected`]: isSelected,
    },
    classNames,
  );

  return (
    <div {...props} className={cnTag} ref={ref}>
      <Text as="span" text={text} textSize="medium" onClick={onSelected} />
      {onDismissible && (
        <Icon name="action_close" color="N700" classNames={[`${classTag}-icon`]} size="small" onClick={onDismissible} />
      )}
    </div>
  );
};

export const Tag = forwardRef<HTMLDivElement, TagProps>(TagComponent);
