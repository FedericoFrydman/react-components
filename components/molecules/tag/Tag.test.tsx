import { shallow, ShallowWrapper } from 'enzyme';
import { Tag } from './Tag';

describe('Tag Component', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Tag text="something" />);
  });

  test('should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
