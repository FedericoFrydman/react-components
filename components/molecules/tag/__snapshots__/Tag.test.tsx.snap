// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Component should render the component 1`] = `
<div
  className="lsux-tag"
>
  <ForwardRef(TextComponent)
    as="span"
    text="something"
    textSize="medium"
  />
</div>
`;
